sklearn/cross_decomposition/pls_.py

```diff
@@ -56,7 +57,9 @@ def _nipals_twoblocks_inner_loop(X, Y, mode="A", max_iter=500, tol=1e-06,
             y_weights = np.dot(Y.T, x_score) / np.dot(x_score.T, x_score)
         ## 2.2 Normalize y_weights
         if norm_y_weights:
-            y_weights /= np.sqrt(np.dot(y_weights.T, y_weights))
+            y_weights /= np.sqrt(np.dot(y_weights.T, y_weights)) + eps
+        else:
+            y_weights += eps
         # 2.3 Update y_score: the Y latent scores
         y_score = np.dot(Y, y_weights) / np.dot(y_weights.T, y_weights)
```

Comment:
I think it's more correct to do:

```python
        if norm_y_weights:
           y_weights /= np.sqrt(np.dot(y_weights.T, y_weights)) + eps

        # 2.3 Update y_score: the Y latent scores
        y_score = np.dot(Y, y_weights) / (np.dot(y_weights.T, y_weights) + eps)
```
Otherwise we change `y_weights` in the numerator.